PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/README.md

```diff
@@ -16,10 +16,28 @@ The Java Code Quality Analyzer, is a plugin designed to improve the quality of J
+- **Issue**: This can lead to OutOfMemoryErrors, especially with large files or high-volume uploads.
+- **Severity: INFO**
+- **Recommendation**: Use APIs that take a length parameter. Please refer to the [Azure SDK for Java documentation](https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blob-upload-java) for additional information.
+
+
+2. #### Use ServiceBusProcessorClient instead of ServiceBusReceiverAsyncClient.
 - **Anti-pattern**: The use of the Reactor receiver, specifically the `ServiceBusReceiverAsyncClient`, is an anti-pattern. This is because it's a low-level API that provides fine-grained control over message handling. While this might seem beneficial, it requires a high level of proficiency in Reactive programming and is mainly useful when building a Reactive library or an end-to-end Reactive application.
 
 - **Issue**: The main issue with using `ServiceBusReceiverAsyncClient` is its complexity and the requirement for a deep understanding of Reactive programming. This can make it difficult to use correctly and efficiently, especially for developers who are not familiar with Reactive programming paradigms.
-- **Severity: INFO**
+- **Severity: WARNING**
 - **Recommendation**: Instead of using the low-level `ServiceBusReceiverAsyncClient`, it's recommended to use the `ServiceBusProcessorClient`. The `ServiceBusProcessorClient` is a higher-level abstraction that simplifies message consumption. It's designed for most common use cases and should be the primary choice for consuming messages. This makes it a more suitable option for most developers and scenarios. 
-Please refer to the [Azure SDK for Java documentation](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/servicebus/azure-messaging-servicebus/README.md#when-to-use-servicebusprocessorclient) for additional information.
\ No newline at end of file
+Please refer to the [Azure SDK for Java documentation](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/servicebus/azure-messaging-servicebus/README.md#when-to-use-servicebusprocessorclient) for additional information.
+
+3. #### Use sync client operation if usage of block() on an async client.
+- **Anti-Pattern**: Using `block()` on an async client. This practice turns an asynchronous operation into a synchronous one.
+
+- **Issue**: The use of `block()` goes against the non-blocking nature of reactive streams. 
+It can lead to performance issues because it blocks one of the few available threads.
+In reactive applications, avoiding blocking operations is crucial for scalability and responsiveness.
+- **Severity Level: WARNING**
+- **Recommendation**: If you find yourself frequently using `block()` in your code, consider switching to the sync client. 
+The sync client performs operations synchronously without requiring `block()`. 
+Using the sync client can make your code more straightforward and easier to understand.
```

Comment:
Only relevant link I could find on synchronous clients was https://learn.microsoft.com/en-us/azure/developer/java/sdk/async-programming which recommends using async clients. Please let me know if you have any I can paste here

Reply:
@srnagar